hamilton/plugins/pandas_extensions.py

```diff
@@ -126,20 +126,91 @@ def name(cls) -> str:
-    data loading functionality that pandas does. We will be adding this in over time, but for now
-    you can subclass this or open up an issue if this doesn't have what you want."""
+class PandasParquetReader(DataLoader):
+    """Class that handles saving parquet files with pandas.
+    Maps to https://pandas.pydata.org/docs/reference/api/pandas.read_parquet.html#pandas.read_parquet
+    """
 
-    path: str
+    path: Union[str, Path, BytesIO, BufferedReader]
+    # kwargs
+    engine: Literal["auto", "pyarrow", "fastparquet"] = "auto"
+    columns: Optional[List[str]] = None
+    storage_options: Optional[Dict[str, Any]] = None
+    use_nullable_dtypes: bool = False
+    dtype_backend: Literal["numpy_nullable", "pyarrow"] = "numpy_nullable"
+    filesystem: Optional[str] = None
+    filters: Optional[Union[List[Tuple], List[List[Tuple]]]] = None
 
-    def load_data(self, type_: Type[DATAFRAME_TYPE]) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
-        df = pd.read_parquet(self.path)
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        return [DATAFRAME_TYPE]
+
+    def _get_loading_kwargs(self):
+        kwargs = {}
+        if self.engine is not None:
+            kwargs["engine"] = self.engine
+        if self.columns is not None:
+            kwargs["columns"] = self.columns
+        if self.storage_options is not None:
+            kwargs["storage_options"] = self.storage_options
+        if self.use_nullable_dtypes is not None:
+            kwargs["use_nullable_dtypes"] = self.use_nullable_dtypes
+        if self.dtype_backend is not None:
+            kwargs["dtype_backend"] = self.dtype_backend
+        if self.filesystem is not None:
+            kwargs["filesystem"] = self.filesystem
+        if self.filters is not None:
+            kwargs["filters"] = self.filters
+
+        return kwargs
+
+    def load_data(self, type_: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
+        # Loads the data and returns the df and metadata of the pickle
+        df = pd.read_parquet(self.path, **self._get_loading_kwargs())
         metadata = utils.get_file_metadata(self.path)
+
         return df, metadata
 
+    @classmethod
+    def name(cls) -> str:
+        return "parquet"
+
+
+@dataclasses.dataclass
+class PandasParquetWriter(DataSaver):
+    """Class that handles saving parquet files with pandas.
+    Maps to https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_parquet.html#pandas.DataFrame.to_parquet
+    """
+
+    path: Union[str, Path, BytesIO, BufferedReader]
+    # kwargs
+    engine: Literal["auto", "pyarrow", "fastparquet"] = "auto"
+    compression: Optional[str] = "snappy"
+    index: Optional[bool] = None
+    partition_cols: Optional[List[str]] = None
+    storage_options: Optional[Dict[str, Any]] = None
+
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        return [DATAFRAME_TYPE]
+
+    def _get_saving_kwargs(self) -> Dict[str, Any]:
```

Comment:
I need guidance on how to implement the **kwargs that are not listed in the Pandas [docs](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_parquet.html#pandas.DataFrame.to_parquet) nor in the [user guide](https://pandas.pydata.org/docs/user_guide/io.html#parquet)

Reply:
yep. So, options:

1. Determine what those kwargs are and surface them as fields in the class.
2. add a `extra_kwargs` dictionary to the class so people can explicitly pass them in, and then add it to the kwargs argument.

I think (2) is probably simpler.

```python
extra_kwargs: Optional[Dict[str, Any]] = None

...

if self.extra_kwargs is not None:
   kwargs.update(self.extra_kwargs)
```

Reply:
Thanks, that is a simpler option. Although should we handle any errors when a user enters an unexpected keyword argument?

Reply:
No I think that's on them unless it's something really common. We can always add this later.